gocourse02/main.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import "fmt"
+
+// Struct that represents an Animal
+type Animal struct {
+	name      string // Name of the animal
+	species   string // Species of the animal
+	inTheCage bool   // Flag indicating whether the animal is in a cage
+}
+
+// Struct that represents a Zoo, containing multiple animals
+type Zoo struct {
+	Animals []Animal // Slice of animals in the zoo
+}
+
+// Struct that represents a Cage, which can hold an animal
+type Cage struct {
+	name   string // Name of the cage
+	animal string // Species of the animal currently in the cage
+}
+
+// Struct that represents all the cages in the zoo
+type AllCages struct {
+	Cages []Cage // Slice of cages
+}
+
+// Struct that represents a Zookeeper
+type Zookeeper struct {
+	name string // Name of the zookeeper
+}
+
+// Constructor for creating a new Animal
+func NewAnimal(species string, name string) *Animal {
+	return &Animal{species: species, name: name}
+}
+
+// Constructor for creating a new Cage
+func NewCage(name string) *Cage {
+	return &Cage{name: name}
+}
+
+// Constructor for creating a new Zookeeper
+func NewZookeeper() *Zookeeper {
+	return &Zookeeper{name: "Bob"}
+}
+
+// Method that describes the current state of the cage
+func (c Cage) DescribeCage() {
+	if c.animal == "" {
+		fmt.Println("Cage is empty!\n") // Print if the cage is empty
```

Comment:
Ось ця інструкція додасть два переноси рядка. Суфікс `ln` один + `\n` ще один.  

Я думаю, що тут малось на увазі:

```suggestion
		fmt.Println("Cage is empty!")
```

або

```suggestion
		fmt.Printf("Cage is empty!\n")
```

Якщо дійсно треба два переноси рядка, то краще ось так:

```go
		fmt.Printf("Cage is empty!\n\n")
```

Reply:
Коментар `Print if the cage is empty` не потрібен, краще видалити.

Reply:
Видалив і комент і \n